src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -49,6 +50,40 @@ public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Tra
                         .SetMinimumLevel(RazorLSPOptions.GetLogLevelForTrace(trace)))
                     .OnInitialized(async (s, request, response) =>
                     {
+                        bool objectSupportedVersion;
+
+                        if (Version.TryParse(request.ClientInfo?.Version, out var parsedVersion))
+                        {
+                            version = parsedVersion;
+                        }
+
+                        if (version is null)
+                        {
+                            version = new Version("1337");
+                        }
+
+                        if (version > new Version(3, 14, 0, 0))
```

Comment:
I'd switch up these statements a little bit and create a `private static readonly Version Protocol_3_14` 

```C#
if (clientProtocol <= Protocol_3_14)
{
    // Assume any protocol before 3.14 expects a bool capabilities for some of the server side capabilities.

	if(s.ServerSettings.Capabilities.HoverProvider.IsValue)
	{
		s.ServerSettings.Capabilities.HoverProvider = new BooleanOr<HoverOptions>(true);
	}

	if(s.ServerSettings.Capabilities.DocumentRangeFormattingProvider.IsValue)
	{
		s.ServerSettings.Capabilities.DocumentRangeFormattingProvider = new BooleanOr<DocumentRangeFormattingOptions>(true);
	}
}
```

Reply:
Food for thought: If all of this code ends up growing outside of this PR we may want to take the approach of building some sort of protocol configuration system